1. Выполнение домашнего задания по лекциям №23 и №24

IlyatTsk opened the pull request.

IlyatTsk committed:
1. Выполнение домашнего задания по лекциям №23 и №24

shalyaa commented on src/test/java/tests/DuckActionsFlyTests.java:
```diff
@@ -0,0 +1,37 @@
+import org.testng.annotations.Optional;
+import org.testng.annotations.Test;
+
+public class DuckActionsFlyTests extends DuckActionsClient {
```
могу придраться к названию классов тестовых? можно из них убрать Actions и Controller, чтобы названия были короче и было сразу понятно, что тест делает

shalyaa commented on src/test/java/tests/DuckActionsFlyTests.java:
```diff
@@ -0,0 +1,37 @@
+    @CitrusTest
+    @Test(description = "Проверка того, что уточка не полетела")
+    public void notSuccessfulFly(@Optional @CitrusResource TestCaseRunner runner) {
+        duckCreate(runner, "getDuckPropertiesTest/createYellowRubberActiveDuck.json");
```
ты в тестах создаешь одинаковых уточек, судя по названию json, но ждешь разный результат, как так?)

shalyaa commented on src/test/java/tests/DuckActionsPropertiesTests.java:
```diff
@@ -0,0 +1,36 @@
+        extractId(runner);
+        duckShowProperties(runner, "${duckId}");
+
+        String expectedFile = "PropertiesTests/successfulResponseWoodenDuck.json";
```
если очень хочется вынести ссылку на файл, то лучше тогда переменую объявлять в начале теста, чтобы сам тест не резать
когда шаги теста идут друг за другом, то это гораздо читабельнее, и не забудь после объявления переменной сделать перенос строки два раза

shalyaa commented on src/test/java/tests/DuckControllerCreateTests.java:
```diff
@@ -0,0 +1,89 @@
+    @Test(description = "Проверка корректно созданной уточки")
+    public void successfulCreate(@Optional @CitrusResource TestCaseRunner runner) {
+        duckCreate(runner, "getDuckPropertiesTest/createYellowRubberActiveDuck.json");
+        extractId(runner);
```
классно, что проверяешь, что тебе возвращаются верные параметры только что созданной утки, но еще стоит добавить проверку, что на метод create тебе пришло ок и тело ответа ожидаемое

shalyaa commented on src/test/java/tests/DuckControllerDeleteTests.java:
```diff
@@ -0,0 +1,23 @@
+public class DuckControllerDeleteTests extends DuckCRUDClient {
+
+    @CitrusTest
+    @Test(description = "Проверка корректного запроса метода, при создании трех уточек")
```
а где в тесте 3 уточки? описание и тест кажется не соответствуют друг другу

shalyaa reviewed (commented):
Поправь мои замечания и сливай в мастер) в целом очень достойно получилось, есть конечно же мелкие замечания, но они не очень критичные

IlyatTsk committed:
1.Изменил названия тестовых классов
2.Поправил ссылку на правильный json в тесте FlyTests
3.Убрал переменную с ссылкой на json в DuckPropertiesTests
4.Добавил проверку метода create
5.Поправил описание теста Delete
